accounting/src/delta/delta_data_collection/tests/mod.rs

```diff
@@ -15,308 +15,123 @@
-    //         None   - Modify
-    //         None   - Delete
-
-    let mut collection1 = DeltaDataCollection::from_iter(
-        [
-            (1, DataDelta::Create('a')),
-            (2, DataDelta::Create('b')),
-            (3, DataDelta::Modify('a', 'c')),
-            (4, DataDelta::Modify('a', 'd')),
-            (5, DataDelta::Delete('e')),
-            (6, DataDelta::Create('f')),
-            (7, DataDelta::Modify('a', 'g')),
-            (8, DataDelta::Delete('h')),
-        ]
-        .into_iter(),
-    );
-
-    let collection2 = DeltaDataCollection::from_iter(
-        [
-            (1, DataDelta::Modify('a', 'i')),
-            (2, DataDelta::Delete('b')),
-            (3, DataDelta::Modify('c', 'j')),
-            (4, DataDelta::Delete('d')),
-            (5, DataDelta::Create('e')),
-            (9, DataDelta::Create('m')),
-            (10, DataDelta::Modify('a', 'n')),
-            (11, DataDelta::Delete('o')),
-        ]
-        .into_iter(),
-    );
-
-    let expected_data_after_merge = BTreeMap::from_iter(
-        [
-            (1, DeltaMapElement::Delta(DataDelta::Create('i'))),
-            // 2 is No-op
-            (3, DeltaMapElement::Delta(DataDelta::Modify('a', 'j'))),
-            (4, DeltaMapElement::Delta(DataDelta::Delete('a'))),
-            // 5 is No-op
-            (6, DeltaMapElement::Delta(DataDelta::Create('f'))),
-            (7, DeltaMapElement::Delta(DataDelta::Modify('a', 'g'))),
-            (8, DeltaMapElement::Delta(DataDelta::Delete('h'))),
-            (9, DeltaMapElement::Delta(DataDelta::Create('m'))),
-            (10, DeltaMapElement::Delta(DataDelta::Modify('a', 'n'))),
-            (11, DeltaMapElement::Delta(DataDelta::Delete('o'))),
-        ]
-        .into_iter(),
-    );
-
-    let undo_data = collection1.merge_delta_data(collection2).unwrap();
-    assert_eq!(collection1.data, expected_data_after_merge);
-
-    let expected_data_after_undo = BTreeMap::from_iter(
-        [
-            (1, DeltaMapElement::Delta(DataDelta::Create('a'))),
-            (
-                2,
-                DeltaMapElement::DeltaUndo(DataDeltaUndo(DataDelta::Create('b'))),
-            ),
-            (3, DeltaMapElement::Delta(DataDelta::Modify('a', 'c'))),
-            (4, DeltaMapElement::Delta(DataDelta::Modify('a', 'd'))),
-            (
-                5,
-                DeltaMapElement::DeltaUndo(DataDeltaUndo(DataDelta::Delete('e'))),
-            ),
-            (6, DeltaMapElement::Delta(DataDelta::Create('f'))),
-            (7, DeltaMapElement::Delta(DataDelta::Modify('a', 'g'))),
-            (8, DeltaMapElement::Delta(DataDelta::Delete('h'))),
-        ]
-        .into_iter(),
-    );
-    collection1.undo_merge_delta_data(undo_data).unwrap();
-    assert_eq!(collection1.data, expected_data_after_undo);
+#[rstest]
+#[case(DataDelta::new(None, None))]
+#[case(DataDelta::new(None, Some('a')))]
+#[case(DataDelta::new(Some('a'), None))]
+#[case(DataDelta::new(Some('a'), Some('b')))]
```

Comment:
> I'm thinking whether we should just convert `(Some('x'), Some('x'))` to some kind of Noop. The reason is specifically to save memory and kind of an optimization. Does that make sense?

`(Some(x), Some(x))` is not equivalent to a noop. It also checks the value is `'x'` and gives an error otherwise. The proposed optimisation would violate the property that applying deltas one after another gives the same result as combining deltas and applying the result.